cocos/platform/CCFileUtils.cpp

```diff
@@ -569,6 +569,13 @@ bool FileUtils::writeStringToFile(const std::string& dataStr, const std::string&
     return rv;
 }
 
+void FileUtils::writeStringToFile(std::string dataStr, const std::string& fullPath, std::function<void(bool)> callback)
+{
+    performOperationOffthread([fullPath](const std::string& dataStr) -> bool {
+            return FileUtils::getInstance()->writeStringToFile(dataStr, fullPath);
+    }, std::move(callback),std::move(dataStr));
```

Comment:
It is leveraging move sematics to avoid data copying. If I did not use std::move here, it would invoke the copy constructor of callback and dataStr, causing a potentially costly copy operation. Instead it will invoke the move constructor, which is a less expensive move operation (usually just a pointer assignment). 

Here is an example to demonstrate. 

```
class NoCopy {
public:
    NoCopy() = default;
    NoCopy(const NoCopy&) = delete;
    NoCopy& operator=(const NoCopy&) = delete;
    
    NoCopy(NoCopy&&) { CCLOG("IM MOVING!"); }
    NoCopy& operator=(NoCopy&&) { CCLOG("OPERATOR="); return *this; }
};

void moveSemantics(NoCopy value)
{
    NoCopy value2 = std::move(value); //value2 will trigger move constructor of value
    NoCopy copy = value; // compile error, attempts to trigger copy constructor. would
    // copy if copy constructor was available. 
}

int main(int argc, char** argv)
{
    moveSemantics(NoCopy()); // Works fine, NoCopy object is not copied
    NoCopy t;
    moveSemantics(t); // Compile error, causes a copy, would call copy constructor if provided
}
```

In general, if we are assigning values like this, we should be taking them by value instead of by const reference (There are some exceptions to this). Assigning from a cost l-value reference will trigger the copy constructor, while passing by value will allow us to avoid the copy if the callsite is an r-value. Then we can call std::move to trigger the move and avoid copying within the function.

Properly leveraging move semantics gives us the ability to avoid copying, while having the safety given by RAII.